src/coreclr/vm/gcenv.ee.common.cpp

```diff
@@ -149,6 +171,11 @@ void GcEnumObject(LPVOID pData, OBJECTREF *pObj, uint32_t flags)
     Object ** ppObj = (Object **)pObj;
     GCCONTEXT   * pCtx  = (GCCONTEXT *) pData;
 
+    if ((GCStress<cfg_any>::IsEnabled() || g_pConfig->GetEnableGCHoleMonitoring()) && ((flags & GC_CALL_PINNED) == 0) && pCtx->sc->promotion)
```

Comment:
```suggestion
    if (g_pConfig->GetEnableGCHoleMonitoring() && ((flags & GC_CALL_PINNED) == 0) && pCtx->sc->promotion)
```
Instead of checking two conditions here, we can set the default value for `EnableGCHoleMonitoring` in eeconfig.cpp to 1 when GC stress is enabled. It will give us more flexibility by allowing `GCHoleMonitoring` to be disable even when GC stress is enabled.